### PR TITLE
offer result vows in smart wallet 

### LINKED ISSUE #9308: add support for Vows to smart-wallet
## What is the Problem Being Solved?

Currently when an OfferResult is a promise, smartWallet awaits it with `watchPromise` to record the offer result. If, as in the case of an async-flow from orchestration, the result is a Vow for a primitive or for invitationMakers, the current smart wallet implementation will discard the Vow rather than handling what it fulfills to.

https://github.com/Agoric/agoric-sdk/blob/fbd8633ae9f8420a589dd9bc32925418f2dde060/packages/smart-wallet/src/offerWatcher.js#L30-L40

## Description of the Design

We could use `watch` in VowTools in place of the current watchPromises.

We might want to go further and use asyncFlow to refactor much of the smart wallet state handling.

### PR DESCRIPTION
**Pull request by turadg**
closes: #9308

## Description
Makes the offer result watcher able to handle offer results that are vows or promises for vows.

### Security Considerations
none

### Scaling Considerations
none

### Documentation Considerations
none

### Testing Considerations
adds a test for a vow resolving after its source vat restarts

### Upgrade Considerations
This requires an upgrade of the smart-wallet contract, which will have to be deployed in a core-eval.

adds a smart wallet contract upgrade test

### CODE REVIEW

File: packages/smart-wallet/src/offerWatcher.js [hunk 1]
```diff
@@ -274,6 +284,8 @@ export const prepareOfferWatcher = baggage => {
           } else {
             facets.helper.handleError(reason);
           }
+          // throw so the vow watcher propagates the rejection
+          throw reason;
```
Comment (0xpatrickdev) [inline packages/smart-wallet/src/offerWatcher.js:288]: Curious how rejections were propagated before - this change seems unrelated to the rest

Comment (turadg) [reply packages/smart-wallet/src/offerWatcher.js:288]: the function returned the promise, but now that's wrapped in vow watchers

File: packages/boot/test/bootstrapTests/vow-offer-results.test.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    // narrow the type
+    assert(statusRecord.updated === 'offerStatus');
+    t.false('result' in statusRecord.status, 'no result yet');
+  }
+
+  // FIXME fails if valueVow is restarted
+  // t.log('restart valueVow');
+  // await evalProposal(
+  //   buildProposal('@agoric/builders/scripts/testing/restart-valueVow.js'),
+  // );
```
Comment (michaelfig) [inline packages/boot/test/bootstrapTests/vow-offer-results.test.ts]: Did you intend to leave this commented out?

**Review by michaelfig — APPROVED** [review]
Please address my question about the commented-out block.  Otherwise, LGTM!

File: packages/boot/test/bootstrapTests/vow-offer-results.test.ts (hunk 2 above)
Comment (turadg) [reply packages/boot/test/bootstrapTests/vow-offer-results.test.ts]: Oops, bad rebase. This must be tested. Will fixup
